tpm2/structures.go

```diff
@@ -1491,7 +1506,21 @@ func NewTPMUSymMode[C SymModeContents](selector TPMAlgID, contents C) TPMUSymMod
 	}
 }
 
+// Sym returns the 'sym' member of the union.
+func (u *TPMUSymMode) Sym() (*TPMIAlgSymMode, error) {
+	switch u.selector {
+	case TPMAlgTDES, TPMAlgAES, TPMAlgSM4, TPMAlgCamellia:
+		value := u.contents.(*boxed[TPMIAlgSymMode]).unbox()
+		return value, nil
+	default:
+		return nil, fmt.Errorf("did not contain sym (selector value was %v)", u.selector)
+	}
+}
+
 // AES returns the 'aes' member of the union.
+//
+// Deprecated: AES exists for historical compatibility
```

Comment:
The structure does indeed have members for each algorithm, see https://trustedcomputinggroup.org/wp-content/uploads/TPM-2.0-1.83-Part-2-Structures.pdf#page=150 (and see https://trustedcomputinggroup.org/wp-content/uploads/TPM-2.0-1.83-Part-2-Structures.pdf#page=32 for an explanation of the algorithm tokens). `sym` is there as a convenience within the reference implementation and IMHO probably shouldn't even be mentioned in the spec. Also, I'm expecting that a future version of the spec flattens these macros, because they're hard to understand.

Reply:
Oh! I really didn't understand the specification like that the first time I look at it (it was really not clear), but looking at it a second time, I actually understand.